Add reindexing doc (#2393)
ordinals/ord: docs/src/guides/reindexing.md

Reindexing
==========

> Sometimes the `ord` database has to be reindexed, which means deleting the

Reviewer: ```suggestion
Sometimes the `ord` database must be reindexed, which means deleting the
```

database and restarting the indexing process with either `ord index run` or
`ord server`. Reasons to reindex are:

1. A new major release of ord, which changes the database scheme
2. The database got corrupted somehow

The database `ord` uses is called [redb](https://github.com/cberner/redb),

> which is an embedded database with the default file name `index.redb`. By

Reviewer: This kinda makes it sound like `index.redb` is redb's default filename.

default we store this file in different locations depending on your operating
system.

|Platform | Value                                            | Example                                      |
| ------- | ------------------------------------------------ | -------------------------------------------- |
| Linux   | `$XDG_DATA_HOME`/ord or `$HOME`/.local/share/ord | /home/alice/.local/share/ord                 |
| macOS   | `$HOME`/Library/Application Support/ord          | /Users/Alice/Library/Application Support/ord |
| Windows | `{FOLDERID_RoamingAppData}`\ord                  | C:\Users\Alice\AppData\Roaming\ord           |

So to delete the database and reindex on MacOS you would have to run the following
commands in the terminal:

```bash

> rm /Users/Alice/Library/Application Support/ord/index.redb

Reviewer: ```suggestion
rm ~/Library/Application Support/ord/index.redb
```
